index.bs

```diff
@@ -1370,77 +896,768 @@ What configuration is the device in after it resets?
+            1. <a>Abort transfers currently scheduled on an interface</a> with |interface|.
+    1. Issue a <code>SET_CONFIGURATION</code> <a>control transfer</a> with
+        <code>configurationValue</code> set to the |configurationValue|.
+        If this step fails <a>reject</a> |promise| with a {{NetworkError}} and
+        abort these steps.
+    1. Let |numInterfaces| be the <a>size</a> of |selectedConfiguration|.{{USBConfiguration/[[interfaces]]}}.
+    1. Resize <a>this</a>.{{USBDevice/[[selectedAlternateSetting]]}} to |numInterfaces|.
+    1. Fill <a>this</a>.{{USBDevice/[[selectedAlternateSetting]]}} with 0.
+    1. Resize <a>this</a>.{{USBDevice/[[claimedInterface]]}} to |numInterfaces|.
+    1. Fill <a>this</a>.{{USBDevice/[[claimedInterface]]}} with <code>false</code>.
+    1. Set <a>this</a>.{{USBDevice/[[configurationValue]]}} to |configurationValue| 
+        and <a>resolve</a> |promise|.
+</div>
+
+<div algorithm="USBDevice.claimInterface(interfaceNumber)">
+    The <dfn for=USBDevice method>claimInterface(|interfaceNumber|)</dfn> method,
+    when invoked, MUST return a new {{Promise}} |promise| and run the following 
+    steps <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
+    1. Let |activeConfiguration| be the result of <a>finding the current configuration</a>
+        with <a>this</a>.
+    1. Let |interfaces| be |activeConfiguration|.{{USBConfiguration/[[interfaces]]}}.
+    1. Let |interfaceIndex| be the result of <a>finding the interface index</a> with
+        |interfaceNumber| and |activeConfiguration|.
+    1. If |interfaceIndex| is equal to <code>-1</code>, <a>reject</a> |promise|
+        with a {{NotFoundError}} and abort these steps.
+    1. If <a>this</a>.{{USBDevice/[[claimedInterface]]}}[|interfaceIndex|] is <code>true</code>,
+        <a>resolve</a> |promise| and abort these steps.
+    1. If |interfaces|[|interfaceIndex|].{{USBInterface/[[isProtectedClass]]}} is <code>true</code>,
+        [=reject=] |promise| with a {{SecurityError}} and abort these steps.
+    1. Perform the necessary platform-specific steps to request exclusive control
+        over |interfaces|[|interfaceIndex|] for the current execution context. If this fails,
+        <a>reject</a> |promise| with a {{NetworkError}} and abort these steps.
+    1. Set <a>this</a>.{{USBDevice/[[claimedInterface]]}}[|interfaceIndex|] to <code>true</code>
+        and <a>resolve</a> |promise|.
+</div>
+
+<div algorithm="USBDevice.releaseInterface(interfaceNumber)">
+    The <dfn for=USBDevice method>releaseInterface(|interfaceNumber|)</dfn> method,
+    when invoked, MUST return a new {{Promise}} |promise| and run the following
+    steps <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
```

Comment:
```suggestion
        and abort these steps if |promise| is rejected.
```

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/5aaf696938ed1dc073f8cc8da8c809e9e5019f41.